presto-router/src/main/java/com/facebook/presto/router/spec/RouterSpec.java

```diff
@@ -38,7 +39,12 @@ public RouterSpec(
     {
         this.groups = ImmutableList.copyOf(requireNonNull(groups, "groups is null"));
         this.selectors = ImmutableList.copyOf(requireNonNull(selectors, "selectors is null"));
-        this.schedulerType = requireNonNull(schedulerType, "scheduler is null");
+        if (schedulerType == null) {
```

Comment:
is it possible to use Optional for schedulerType?

Reply:
Nice catch! I updated the type of `schedulerType` to `Optional`.